Upgrade to StoryBook 8 (#5912)
plone/volto: docs/source/upgrade-guide/index.md

import { slateBeforeEach } from '@plone/volto/cypress/support/helpers';
```

### StoryBook 8

StoryBook was upgraded from version 6 to 8 in core and in the project generator.
This section is relevant if you have StoryBook stories in your project or add-on.
The versions will be upgraded automatically using the `volto-update-deps` script.
The configuration of your project must also be updated with the new one.
Replace the `.storybook` folder in your project with this one:

>  `https://github.com/plone/volto/tree/main/packages/generator-volto/generators/app/templates/.storybook`

Reviewer: I think we want to use the permalink for the latest commit, as the main branch may have changes in the future. Then again, maybe that would be a good thing so that the upgrade guide always links to the latest version of this folder? I'm not sure which is better.

In either case, a naked URL will be converted to a clickable link, and is preferable to use an inline literal. Choose one:

```suggestion
https://github.com/plone/volto/tree/main/packages/generator-volto/generators/app/templates/.storybook

=== OR ===

https://github.com/plone/volto/tree/5605131868689778bbdca0c3003a40cb9f153c1a/packages/generator-volto/generators/app/templates/.storybook
```

Author: Added permalink

> You can find more information about the migration in the [official StoryBook docs](https://storybook.js.org/docs/migration-guide/from-older-version).

Reviewer: ````suggestion

```{seealso}
[Migration guide from Storybook 6.x to 8.0](https://storybook.js.org/docs/migration-guide/from-older-version)
```

````

> If you haven't customized the configuration, the migration is quite straightforward.

Reviewer: ```suggestion
If you haven't customized the configuration, the migration is straightforward.
```

The stories format (CSF) is almost the same.

> However, writting stories in MDX and the `.stories.mdx` extensions are no longer supported.

Reviewer: ```suggestion
However, writing stories directly in MDX was removed in Storybook 8.
The `.stories.mdx` extension is no longer supported.
```

```{note}

> Technically is possible to keep the old version running, but the `volto-update-deps` will try to update them every time you run it.

Reviewer: ```suggestion
Although it is technically possible to keep the old version running, the script `volto-update-deps` will try to update to Storybook 8 every time you run it.
```
